Tests/GRPCCoreTests/StatusTests.swift

```diff
@@ -13,105 +13,60 @@
-    for (code, expected) in Self.statusCodeRawValue {
-      XCTAssertEqual(code.rawValue, expected, "\(code) had unexpected raw value")
+@Suite("Status")
+struct StatusTests {
+  @Suite("Code")
+  struct Code {
+    @Test("rawValue", arguments: zip(Status.Code.all, 0 ... 16))
+    func rawValueOfStatusCodes(code: Status.Code, expected: Int) {
+      #expect(code.rawValue == expected)
     }
-  }
-
-  func testStatusCodeFromErrorCode() throws {
-    XCTAssertEqual(Status.Code(RPCError.Code.cancelled), .cancelled)
-    XCTAssertEqual(Status.Code(RPCError.Code.unknown), .unknown)
-    XCTAssertEqual(Status.Code(RPCError.Code.invalidArgument), .invalidArgument)
-    XCTAssertEqual(Status.Code(RPCError.Code.deadlineExceeded), .deadlineExceeded)
-    XCTAssertEqual(Status.Code(RPCError.Code.notFound), .notFound)
-    XCTAssertEqual(Status.Code(RPCError.Code.alreadyExists), .alreadyExists)
-    XCTAssertEqual(Status.Code(RPCError.Code.permissionDenied), .permissionDenied)
-    XCTAssertEqual(Status.Code(RPCError.Code.resourceExhausted), .resourceExhausted)
-    XCTAssertEqual(Status.Code(RPCError.Code.failedPrecondition), .failedPrecondition)
-    XCTAssertEqual(Status.Code(RPCError.Code.aborted), .aborted)
-    XCTAssertEqual(Status.Code(RPCError.Code.outOfRange), .outOfRange)
-    XCTAssertEqual(Status.Code(RPCError.Code.unimplemented), .unimplemented)
-    XCTAssertEqual(Status.Code(RPCError.Code.internalError), .internalError)
-    XCTAssertEqual(Status.Code(RPCError.Code.unavailable), .unavailable)
-    XCTAssertEqual(Status.Code(RPCError.Code.dataLoss), .dataLoss)
-    XCTAssertEqual(Status.Code(RPCError.Code.unauthenticated), .unauthenticated)
-  }
 
-  func testStatusCodeFromValidRawValue() {
-    for (expected, rawValue) in Self.statusCodeRawValue {
-      XCTAssertEqual(
-        Status.Code(rawValue: rawValue),
-        expected,
-        "\(rawValue) didn't convert to expected code \(expected)"
+    @Test(
+      "Initialize from RPCError.Code",
+      arguments: zip(
+        RPCError.Code.all,
+        Status.Code.all.dropFirst()  // Drop '.ok', there is no '.ok' error code.
       )
+    )
+    func initFromRPCErrorCode(errorCode: RPCError.Code, expected: Status.Code) {
+      #expect(Status.Code(errorCode) == expected)
     }
-  }
 
-  func testStatusCodeFromInvalidRawValue() {
-    // Internally represented as a `UInt8`; try all other values.
-    for rawValue in UInt8(17) ... UInt8.max {
-      XCTAssertNil(Status.Code(rawValue: Int(rawValue)))
+    @Test("Initialize from rawValue", arguments: zip(0 ... 16, Status.Code.all))
+    func initFromRawValue(rawValue: Int, expected: Status.Code) {
+      #expect(Status.Code(rawValue: rawValue) == expected)
     }
 
-    // API accepts `Int` so try invalid `Int` values too.
-    XCTAssertNil(Status.Code(rawValue: -1))
-    XCTAssertNil(Status.Code(rawValue: 1000))
-    XCTAssertNil(Status.Code(rawValue: .max))
+    @Test("Initialize from invalid rawValue", arguments: [-1, 17, 100, .max])
```

Comment:
I'm not sure it's really important, but we're not testing for all values from 17 to UInt8.max anymore

Reply:
Yeah, not sure it really matters tbh. The `UInt8` is an implementation detail so a bit odd to do all of those values given the parameter is an `Int`.